### PR TITLE
Modification for attribute precedence issue

### PR DESCRIPTION
**Pull request by Aliasgar16**

### CODE REVIEW

**Comment by Aliasgar16** [comment]
For more details refer https://github.com/chef/knife-ec2/issues/328

File: lib/chef/knife/ec2_server_create.rb [hunk 1]
```diff
@@ -824,7 +824,7 @@ def create_server_def
           :groups => config[:security_groups],
           :security_group_ids => locate_config_value(:security_group_ids),
           :flavor_id => locate_config_value(:flavor),
-          :key_name => Chef::Config[:knife][:aws_ssh_key_id],
+          :key_name => config[:ssh_key_name] || Chef::Config[:knife][:aws_ssh_key_id],
```
Comment (siddheshwar-more) [inline lib/chef/knife/ec2_server_create.rb]: I think we can replace this with `locate_config_value(:ssh_key_name)` . locate_config_value will take care of precedence.

Comment (Aliasgar16) [inline lib/chef/knife/ec2_server_create.rb]: 'locate_config_value' method accesses values from both the configs(CLI and knife) through the same name, but here the issue is that CLI option name in code is 'ssh_key_name', while on the other hand knife option name is 'aws_ssh_key_id', for other options the names are same so it works for them.

Comment (siddheshwar-more) [inline lib/chef/knife/ec2_server_create.rb]: @Aliasgar16 
Ohh Ok. Then,
I think we should change  `:aws_ssh_key_id` option to `:ssh_key_name` (i.e make consistent with others option) but also support `:aws_ssh_key_id` with depreciation warning message, eventually we will remove support for `:aws_ssh_key_id`

@adamedx ^ Please comment on this!

Comment (Aliasgar16) [inline lib/chef/knife/ec2_server_create.rb]: yes...I will do the required changes to make name consistency between :aws_ssh_key_id and :ssh_key_name option, along with deprecation message.

**Pull request merged**